spec/datetimeformat.html

```diff
@@ -260,19 +301,82 @@
+              1. Let _fv_ be an implementation and locale dependent String value representing "ante meridiem".
+            1. Add new part record { [[type]]: "dayperiod", [[value]]: _fv_ } as a new element of the list _result_.
           1. Else,
-            1. Let _fv_ be an implementation and locale dependent String value representing "ante meridiem".
-          1. Replace the substring of _result_ that consists of *"{ampm}"*, with _fv_.
+            1. <emu-note>Unknown token</emu-note>
+            1. Let _separatorValue_ be the substring of _pattern_ from position _beginIndex_, inclusive, to position _endIndex_, inclusive.
+            1. Add new part record { [[type]]: "separtor", [[value]]: _separatorValue_ } as a new element of the list _result_.
+          1. Set _index_ to _endIndex_ + 1.
+          1. Set _beginIndex_ to Call(*%StringProto_indexOf%*, _pattern_, "}", _index_)
+        1. If _endIndex_ is less than _S_, then:
+          1. Let _separator_ to be a new Record.
+          1. Let _separatorValue_ be the substring of _pattern_ from position _endIndex_, exclusive, to position _S_, exclusive.
+          1. Add new part record { [[type]]: "separator", [[value]]: _separatorValue_ } as a new element of the list _result_.
         1. Return _result_.
       </emu-alg>
 
       <emu-note>
+        CreateDateTimeParts abstract operation interpretes _x_ as a time value as specified in ES2015, 20.3.1.1; and create the corresponding parts according to the effective locale and the formatting options of _dateTimeFormat_.
+      </emu-note>
+
+      <emu-note>
         It is recommended that implementations use the locale and calendar dependent strings provided by the Common Locale Data Repository (available at <a href="http://cldr.unicode.org/">http://cldr.unicode.org/</a>), and use CLDR "abbreviated" strings for DateTimeFormat "short" strings, and CLDR "wide" strings for DateTimeFormat "long" strings.
       </emu-note>
+
+      <emu-note>
+        It is recommended that implementations use the time zone information of the IANA Time Zone Database.
+      </emu-note>
+
+    </emu-clause>
+
+    <emu-clause id="alg-FormatDateTime" aoid="DateTimeFormat">
+
+      <h1>FormatDateTime(dateTimeFormat, x)</h1>
+
+      <p>
+        The FormatDateTime abstract operation is called with arguments _dateTimeFormat_ (which must be an object initialized as a DateTimeFormat) and _x_ (which must be a Number value), and performs the following steps:
+      </p>
+
+      <emu-alg>
+        1. Let _parts_ be CreateDateTimeParts(_dateTimneFormat_, _x_).
+        1. Let _result_ be an empty string.
+        1. For each _part_ in _parts_, do:
+          1. Set _result_ to a String value produced by concatenating _result_ and _part_.[[value]].
+        1. Return _result_.
+      </emu-alg>
+
+    </emu-clause>
+
+    <emu-clause id="alg-FormatToPartDateTime" aoid="DateTimeFormatToParts">
+
+      <h1>FormatToPartDateTime(dateTimeFormat, x)</h1>
```

Comment:
Perhaps I'm missing something, but where is this abstract operation used?  The `aoid` seems to be different from the name of the operation.